azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,258 @@
+// Copyright 2016-2019, Pulumi Corporation.  All rights reserved.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+
+import * as msRestAzure from "@azure/ms-rest-nodeauth";
+import * as cdnManagement from "@azure/arm-cdn";
+import { ServiceClientCredentials } from "@azure/ms-rest-js";
+
+/**
+ * CustomDomainOptions represents the inputs to the dynamic resource.
+ * Any property of type `Input<T>` will automatically be resolved by the custom dynamic resource
+ * before passing them to the dynamic resource provider.
+ */
+export interface CustomDomainOptions {
+    resourceGroupName: pulumi.Input<string>;
+    profileName: pulumi.Input<string>;
+    endpointName: pulumi.Input<string>;
+    customDomainHostName: pulumi.Input<string>;
+    httpsEnabled: boolean;
+}
+
+/**
+ * DynamicProviderInputs represents the inputs that are passed as inputs
+ * to each function in the implementation of a `pulumi.dynamic.ResourceProvider`.
+ * The property names in this must match `CustomDomainOptions`.
+ */
+interface DynamicProviderInputs {
```

Comment:
Don't you like my suggested trick with a mapped type?

Reply:
Ah it got buried under my updates. I'll reply to that comment.